### ISSUE TITLE
[Config Support]: Is it possible to rotate my TP-LINK C210?

### ISSUE THREAD

**Issue by BrunoTeixeira1996** [blakeblackshear/frigate#8239]
### Describe the problem you are having

Hi guys,
So I have a TAPO TP-LINK C210 and using the app I can rotate the camera 360º
I was wondering if I can do the same using frigate.

### Version

0.12.1-367d724

### Frigate config file

```yaml
mqtt:
  enabled: false

go2rtc:
  streams:
    front_cam:
      - rtsp://user:pw@192.168.30.44:554/stream1
    front_cam_rotated_90:
      - "ffmpeg:front_cam#video=h264#rotate=90"

cameras:
  front_cam:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/front_cam
          input_args: preset-rtsp-restream
          roles: detect
    live:
         stream_name: front_cam
```

### Relevant log output

```shell
Nothing
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Any other information that may be helpful

The main Idea is to use frigate when I dont have my phone available to check the camera stream but since the camera can rotate 360º I want to do that.
The only thing i've found was rotate the stream image.

**Comment by NickM-27** [comment]
yes, what you have should be good except you are referencing the wrong stream and you should probably use hwaccel if it is available

**Comment by BrunoTeixeira1996** [comment]
> yes, what you have should be good except you are referencing the wrong stream and you should probably use hwaccel if it is available

Can you point me to the right direction? what should I change in my config?

**Comment by NickM-27** [comment]
what hardware are you running frigate on

**Comment by BrunoTeixeira1996** [comment]
> what hardware are you running frigate on

Inside a LXC in Proxmox using Docker container

**Comment by NickM-27** [comment]
that's software, I asked about hardware

**Comment by BrunoTeixeira1996** [comment]
> that's software, I asked about hardware



CPU | 2x Xeon E5-2670 2.60GHz Eight 8-Core
-- | --
RAM | 24 GB RAM 1600MHz Kingston
GPU | ATI Radeon HD-3470 256 MB GDDR2

CPU	[2x Xeon E5-2670 2.60GHz Eight 8-Core](https://ark.intel.com/content/www/us/en/ark/products/series/59138/intel-xeon-processor-e5-family.html)
RAM	24 GB RAM 1600MHz Kingston
GPU	ATI Radeon HD-3470 256 MB GDDR2

**Comment by NickM-27** [comment]
so assuming the GPU is available in the container you should be able to add `#hardware` to the end of the go2rtc stream and enable hwaccel according to the docs https://docs.frigate.video/configuration/hardware_acceleration#amdati-gpus-radeon-hd-2000-and-newer-gpus-via-libva-mesa-driver

**Comment by BrunoTeixeira1996** [comment]
> so assuming the GPU is available in the container you should be able to add `#hardware` to the end of the go2rtc stream and enable hwaccel according to the docs https://docs.frigate.video/configuration/hardware_acceleration#amdati-gpus-radeon-hd-2000-and-newer-gpus-via-libva-mesa-driver

Sorry I didn't get it. Looking at my config file where should I use `#hardware` and where should I enable `hwaccel`?

**Comment by NickM-27** [comment]
```
go2rtc:
  streams:
    front_cam:
      - rtsp://user:pw@192.168.30.44:554/stream1
    front_cam_rotated_90:
      - "ffmpeg:front_cam#video=h264#rotate=90#hardware"
```

will enable hwaccel for the go2rtc stream, and the instructions I linked are for enabling hwaccel in the frigate streams

**Comment by BrunoTeixeira1996** [comment]
> #hardware

Using this config I am having the following error

```
mqtt:
  enabled: false

ffmpeg:
  hwaccel_args: preset-vaapi

go2rtc:
  streams:
    front_cam:
      - rtsp://user:pw@192.168.30.44:554/stream1
    front_cam_rotated_90:
      - "ffmpeg:front_cam#video=h264#rotate=90#hardware"

birdseye:
  enabled: True
  mode: continuous

cameras:
  front_cam:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/front_cam
          input_args: preset-rtsp-restream
          roles: detect
    live:
         stream_name: front_cam
```


```
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service nginx: stopping
s6-rc: info: service go2rtc-healthcheck: stopping
2023-10-19 22:32:07.123798251  [INFO] The go2rtc-healthcheck service exited with code 256 (by signal 15)
s6-rc: info: service go2rtc-healthcheck successfully stopped
2023-10-19 22:32:07.181217291  [INFO] Service NGINX exited with code 0 (by signal 0)
s6-rc: info: service nginx successfully stopped
s6-rc: info: service nginx-log: stopping
s6-rc: info: service frigate: stopping
2023-10-19 22:32:07.185036600  [2023-10-19 22:32:07] frigate.app                    INFO    : Stopping...
s6-rc: info: service nginx-log successfully stopped
2023-10-19 22:32:07.186605011  [2023-10-19 22:32:07] detector.cpu                   INFO    : Signal to exit detection process...
2023-10-19 22:32:07.186608224  [2023-10-19 22:32:07] root                           INFO    : Waiting for detection process to exit gracefully...
2023-10-19 22:32:07.186610073  [2023-10-19 22:32:07] frigate.storage                INFO    : Exiting storage maintainer...
2023-10-19 22:32:07.186611665  [2023-10-19 22:32:07] frigate.events                 INFO    : Exiting event cleanup...
2023-10-19 22:32:07.186613242  [2023-10-19 22:32:07] frigate.record                 INFO    : Exiting recording cleanup...
2023-10-19 22:32:07.186614824  [2023-10-19 22:32:07] frigate.stats                  INFO    : Exiting stats emitter...
2023-10-19 22:32:07.186616407  [2023-10-19 22:32:07] frigate.record                 INFO    : Exiting recording maintenance...
2023-10-19 22:32:07.186617875  [2023-10-19 22:32:07] frigate.watchdog               INFO    : Exiting watchdog...
2023-10-19 22:32:07.501052589  [2023-10-19 22:32:07] detector.cpu                   INFO    : Exited detection process...
2023-10-19 22:32:07.504894027  [2023-10-19 22:32:07] root                           INFO    : Detection process has exited...
2023-10-19 22:32:07.511370964  [2023-10-19 22:32:07] frigate.object_processing      INFO    : Exiting object processor...
2023-10-19 22:32:07.838999222  [2023-10-19 22:32:07] frigate.events                 INFO    : Exiting event processor...
2023-10-19 22:32:08.022652127  [2023-10-19 22:32:08] frigate.comms.ws               INFO    : Exiting websocket client...
2023-10-19 22:32:08.023236110  [2023-10-19 22:32:08] peewee.sqliteq                 INFO    : writer received shutdown request, exiting.
2023-10-19 22:32:08.024538855  [2023-10-19 22:32:08] watchdog.front_cam             INFO    : Terminating the existing ffmpeg process...
2023-10-19 22:32:08.024542229  [2023-10-19 22:32:08] watchdog.front_cam             INFO    : Waiting for ffmpeg to exit gracefully...
2023-10-19 22:32:08.521200672  [2023-10-19 22:32:08] frigate.video                  INFO    : front_cam: exiting subprocess
2023-10-19 22:32:09.037819583  [2023-10-19 22:32:09] frigate.output                 INFO    : exiting output process...
2023-10-19 22:32:10.179711771  [INFO] Service Frigate exited with code 0 (by signal 0)
s6-rc: info: service frigate successfully stopped
s6-rc: info: service go2rtc: stopping
s6-rc: info: service frigate-log: stopping
2023-10-19 22:32:10.182438343  exit OK
s6-rc: info: service frigate-log successfully stopped
2023-10-19 22:32:10.188007103  [INFO] The go2rtc service exited with code 0 (by signal 0)
s6-rc: info: service go2rtc successfully stopped
s6-rc: info: service go2rtc-log: stopping
s6-rc: info: service go2rtc-log successfully stopped
s6-rc: info: service log-prepare: stopping
s6-rc: info: service s6rc-fdholder: stopping
s6-rc: info: service log-prepare successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service s6rc-fdholder successfully stopped
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
s6-rc: info: service s6rc-fdholder: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service s6rc-fdholder successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service log-prepare: starting
s6-rc: info: service log-prepare successfully started
s6-rc: info: service nginx-log: starting
s6-rc: info: service go2rtc-log: starting
s6-rc: info: service frigate-log: starting
s6-rc: info: service nginx-log successfully started
s6-rc: info: service go2rtc-log successfully started
s6-rc: info: service go2rtc: starting
s6-rc: info: service frigate-log successfully started
s6-rc: info: service go2rtc successfully started
s6-rc: info: service go2rtc-healthcheck: starting
s6-rc: info: service frigate: starting
2023-10-19 22:32:14.859641528  [INFO] Preparing go2rtc config...
s6-rc: info: service frigate successfully started
s6-rc: info: service nginx: starting
s6-rc: info: service nginx successfully started
2023-10-19 22:32:14.869961710  [INFO] Starting NGINX...
s6-rc: info: service go2rtc-healthcheck successfully started
2023-10-19 22:32:14.886153268  [INFO] Starting Frigate...
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
2023-10-19 22:32:15.194593096  [INFO] Starting go2rtc...
2023-10-19 22:32:15.300580256  22:32:15.300 INF go2rtc version 1.2.0 linux/amd64
2023-10-19 22:32:15.301566309  22:32:15.301 INF [api] listen addr=:1984
2023-10-19 22:32:15.302081098  22:32:15.302 INF [rtsp] listen addr=:8554
2023-10-19 22:32:15.302564680  22:32:15.302 INF [srtp] listen addr=:8443
2023-10-19 22:32:15.303001479  22:32:15.302 INF [webrtc] listen addr=:8555
2023-10-19 22:32:16.007290919  [2023-10-19 22:32:15] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2023-10-19 22:32:16.017410631  [2023-10-19 22:32:16] peewee_migrate                 INFO    : Starting migrations
2023-10-19 22:32:16.022639024  [2023-10-19 22:32:16] peewee_migrate                 INFO    : There is nothing to migrate
2023-10-19 22:32:16.036538367  [2023-10-19 22:32:16] frigate.app                    INFO    : Output process started: 1208
2023-10-19 22:32:16.049243991  [2023-10-19 22:32:16] detector.cpu                   INFO    : Starting detection process: 1207
2023-10-19 22:32:16.049851569  [2023-10-19 22:32:16] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-10-19 22:32:16.057337482  [2023-10-19 22:32:16] frigate.app                    INFO    : Camera processor started for front_cam: 1213
2023-10-19 22:32:16.057341980  [2023-10-19 22:32:16] frigate.app                    INFO    : Capture process started for front_cam: 1215
2023-10-19 22:32:18.603522128  [2023-10-19 22:32:18] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:32:18.603789413  [2023-10-19 22:32:18] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:32:19.567980128  192.168.30.100 - - [19/Oct/2023:22:32:19 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:24.931662204  [INFO] Starting go2rtc healthcheck service...
2023-10-19 22:32:26.962193545  192.168.30.100 - - [19/Oct/2023:22:32:26 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:27.856624428  192.168.30.100 - - [19/Oct/2023:22:32:27 +0100] "GET / HTTP/1.1" 200 6641 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:28.064782298  192.168.30.100 - - [19/Oct/2023:22:32:28 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:28.224451326  192.168.30.100 - - [19/Oct/2023:22:32:28 +0100] "GET //api/front_cam/latest.jpg?h=353 HTTP/1.1" 200 9382 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
```

**Comment by NickM-27** [comment]
need to wait longer for the actual error to be put into the logs

**Comment by BrunoTeixeira1996** [comment]
> need to wait longer for the actual error to be put into the logs

```
2023-10-19 22:32:15.302081098  22:32:15.302 INF [rtsp] listen addr=:8554
2023-10-19 22:32:15.302564680  22:32:15.302 INF [srtp] listen addr=:8443
2023-10-19 22:32:15.303001479  22:32:15.302 INF [webrtc] listen addr=:8555
2023-10-19 22:32:16.007290919  [2023-10-19 22:32:15] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2023-10-19 22:32:16.017410631  [2023-10-19 22:32:16] peewee_migrate                 INFO    : Starting migrations
2023-10-19 22:32:16.022639024  [2023-10-19 22:32:16] peewee_migrate                 INFO    : There is nothing to migrate
2023-10-19 22:32:16.036538367  [2023-10-19 22:32:16] frigate.app                    INFO    : Output process started: 1208
2023-10-19 22:32:16.049243991  [2023-10-19 22:32:16] detector.cpu                   INFO    : Starting detection process: 1207
2023-10-19 22:32:16.049851569  [2023-10-19 22:32:16] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-10-19 22:32:16.057337482  [2023-10-19 22:32:16] frigate.app                    INFO    : Camera processor started for front_cam: 1213
2023-10-19 22:32:16.057341980  [2023-10-19 22:32:16] frigate.app                    INFO    : Capture process started for front_cam: 1215
2023-10-19 22:32:18.603522128  [2023-10-19 22:32:18] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:32:18.603789413  [2023-10-19 22:32:18] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:32:19.567980128  192.168.30.100 - - [19/Oct/2023:22:32:19 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:24.931662204  [INFO] Starting go2rtc healthcheck service...
2023-10-19 22:32:26.962193545  192.168.30.100 - - [19/Oct/2023:22:32:26 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:27.856624428  192.168.30.100 - - [19/Oct/2023:22:32:27 +0100] "GET / HTTP/1.1" 200 6641 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:28.064782298  192.168.30.100 - - [19/Oct/2023:22:32:28 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:28.224451326  192.168.30.100 - - [19/Oct/2023:22:32:28 +0100] "GET //api/front_cam/latest.jpg?h=353 HTTP/1.1" 200 9382 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:32:36.091906948  [2023-10-19 22:32:36] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:32:36.091913671  [2023-10-19 22:32:36] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:32:36.091919752  [2023-10-19 22:32:36] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:32:36.091922341  [2023-10-19 22:32:36] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:32:36.091924286  [2023-10-19 22:32:36] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:32:38.644886653  [2023-10-19 22:32:38] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:32:38.645421556  [2023-10-19 22:32:38] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:32:46.103200047  [2023-10-19 22:32:46] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:32:46.103206181  [2023-10-19 22:32:46] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:32:46.103208786  [2023-10-19 22:32:46] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:32:46.103210521  [2023-10-19 22:32:46] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:32:46.103212268  [2023-10-19 22:32:46] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:32:48.605410260  [2023-10-19 22:32:48] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:32:48.605623136  [2023-10-19 22:32:48] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:32:56.110577677  [2023-10-19 22:32:56] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:32:56.110584014  [2023-10-19 22:32:56] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:32:56.110586784  [2023-10-19 22:32:56] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:32:56.110588518  [2023-10-19 22:32:56] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:32:56.110590226  [2023-10-19 22:32:56] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:32:58.704347178  [2023-10-19 22:32:58] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:32:58.704727924  [2023-10-19 22:32:58] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:06.113835999  [2023-10-19 22:33:06] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:06.113842663  [2023-10-19 22:33:06] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:06.113845373  [2023-10-19 22:33:06] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:06.113847156  [2023-10-19 22:33:06] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:06.113848840  [2023-10-19 22:33:06] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:08.595382270  [2023-10-19 22:33:08] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:08.595736585  [2023-10-19 22:33:08] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:16.125095422  [2023-10-19 22:33:16] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:16.125101529  [2023-10-19 22:33:16] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:16.125104145  [2023-10-19 22:33:16] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:16.125105874  [2023-10-19 22:33:16] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:16.125107556  [2023-10-19 22:33:16] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:18.703230155  [2023-10-19 22:33:18] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:18.703491266  [2023-10-19 22:33:18] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:26.132398747  [2023-10-19 22:33:26] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:26.132404812  [2023-10-19 22:33:26] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:26.132407430  [2023-10-19 22:33:26] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:26.132409177  [2023-10-19 22:33:26] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:26.132410859  [2023-10-19 22:33:26] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:26.272478763  [2023-10-19 22:33:26] frigate.util                   ERROR   : Unable to poll intel GPU stats: No device filter specified and no discrete/integrated i915 devices found
2023-10-19 22:33:26.272484213  
2023-10-19 22:33:28.157949473  192.168.30.100 - - [19/Oct/2023:22:33:28 +0100] "GET /ws HTTP/1.1" 101 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
2023-10-19 22:33:28.695495172  [2023-10-19 22:33:28] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:28.695811361  [2023-10-19 22:33:28] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:36.139653029  [2023-10-19 22:33:36] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:36.139659203  [2023-10-19 22:33:36] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:36.139661831  [2023-10-19 22:33:36] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:36.139663595  [2023-10-19 22:33:36] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:36.139665272  [2023-10-19 22:33:36] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:38.702949156  [2023-10-19 22:33:38] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:38.703600484  [2023-10-19 22:33:38] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:46.142881415  [2023-10-19 22:33:46] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:46.142887517  [2023-10-19 22:33:46] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:46.142890097  [2023-10-19 22:33:46] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:46.142891845  [2023-10-19 22:33:46] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:46.142893641  [2023-10-19 22:33:46] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:48.701535852  [2023-10-19 22:33:48] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:48.701542101  [2023-10-19 22:33:48] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:33:56.150427329  [2023-10-19 22:33:56] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:33:56.150434050  [2023-10-19 22:33:56] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:33:56.150436696  [2023-10-19 22:33:56] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:33:56.150438442  [2023-10-19 22:33:56] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:33:56.150440177  [2023-10-19 22:33:56] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:33:58.652381476  [2023-10-19 22:33:58] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:33:58.653029506  [2023-10-19 22:33:58] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:34:06.161699150  [2023-10-19 22:34:06] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:34:06.161705427  [2023-10-19 22:34:06] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:34:06.161707983  [2023-10-19 22:34:06] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:34:06.161709730  [2023-10-19 22:34:06] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:34:06.161711408  [2023-10-19 22:34:06] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:34:08.700139814  [2023-10-19 22:34:08] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:34:08.700538730  [2023-10-19 22:34:08] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:34:16.168990193  [2023-10-19 22:34:16] watchdog.front_cam             ERROR   : Ffmpeg process crashed unexpectedly for front_cam.
2023-10-19 22:34:16.168996366  [2023-10-19 22:34:16] watchdog.front_cam             ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-10-19 22:34:16.168998940  [2023-10-19 22:34:16] ffmpeg.front_cam.detect        ERROR   : Option hwaccel (use HW accelerated decoding) cannot be applied to output url vaapi -- you are trying to apply an input option to an output file or vice versa. Move this option before the file it belongs to.
2023-10-19 22:34:16.169000669  [2023-10-19 22:34:16] ffmpeg.front_cam.detect        ERROR   : Error parsing options for output file vaapi.
2023-10-19 22:34:16.169002358  [2023-10-19 22:34:16] ffmpeg.front_cam.detect        ERROR   : Error opening output files: Invalid argument
2023-10-19 22:34:18.801292456  [2023-10-19 22:34:18] frigate.video                  ERROR   : front_cam: Unable to read frames from ffmpeg process.
2023-10-19 22:34:18.801654298  [2023-10-19 22:34:18] frigate.video                  ERROR   : front_cam: ffmpeg process is not running. exiting capture thread...
2023-10-19 22:34:18.849645876  192.168.30.100 - - [19/Oct/2023:22:34:18 +0100] "GET /api/config HTTP/1.1" 200 4702 "http://192.168.30.96:5000/" "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0" "-"
```

**Comment by NickM-27** [comment]
it looks like the LIBVA_DRIVER_NAME env variable has not been set like the docs specified

**Comment by BrunoTeixeira1996** [comment]
> it looks like the LIBVA_DRIVER_NAME env variable has not been set like the docs specified
it is set 
```
root@docker-frigate:/opt/frigate# echo $LIBVA_DRIVER_NAME=radeonsi
radeonsi=radeonsi
```

**Comment by NickM-27** [comment]
okay, then it seems the GPU itself is not being mapped to the container

**Comment by BrunoTeixeira1996** [comment]
> okay, then it seems the GPU itself is not being mapped to the container

btw, do i need that to change the camera orientation?

**Comment by NickM-27** [comment]
also you still have `- path: rtsp://127.0.0.1:8554/front_cam` but if you want to use the rotated stream in frigate then you need to use `- path: rtsp://127.0.0.1:8554/front_cam_rotated_90`

**Comment by BrunoTeixeira1996** [comment]
> also you still have `- path: rtsp://127.0.0.1:8554/front_cam` but if you want to use the rotated stream in frigate then you need to use `- path: rtsp://127.0.0.1:8554/front_cam_rotated_90`

ok but that is the rotated ffpmeg. i wanted to move the camera using frigate.  i dont think thats possible rigth?

**Comment by NickM-27** [comment]
I don't know what you mean, rotating the camera stream using go2rtc+ffmpeg is the most efficient way to do that so it is rotated for both live view and for frigate itself.

**Comment by BrunoTeixeira1996** [comment]
> I don't know what you mean, rotating the camera stream using go2rtc+ffmpeg is the most efficient way to do that so it is rotated for both live view and for frigate itself.

in the app i can rotate the cam 360º so i can check 360º by using the move option.
Using this rotation with ffpmpeg rotates the image, not the camera itself. I want to rotate the camera

**Comment by NickM-27** [comment]
oh, you are referring to PTZ then, frigate 0.13 (currently in beta) supports this https://deploy-preview-6262--frigate-docs.netlify.app/configuration/cameras#setting-up-camera-ptz-controls

**Comment by NickM-27** [comment]
actually, looks like users have found that the tapo c210 specifically has incorrect implementation of this feature in ONVIF and does not work https://deploy-preview-6262--frigate-docs.netlify.app/configuration/cameras#onvif-ptz-camera-recommendations

**Comment by BrunoTeixeira1996** [comment]
> actually, looks like users have found that the tapo c210 specifically has incorrect implementation of this feature in ONVIF and does not work https://deploy-preview-6262--frigate-docs.netlify.app/configuration/cameras#onvif-ptz-camera-recommendations

damn ... so i cant use this feature :(

**Comment by NickM-27** [comment]
not until TP-Link fixes it

**Comment by NickM-27** [comment]
here is where that discussion took place by the way https://github.com/blakeblackshear/frigate/discussions/7799#discussioncomment-7017809

I am going to close this issue as there is currently nothing to be done, feel free to create a new issue if something else comes up
